Tests/Controller/AbstractControllerTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Payum\Bundle\PayumBundle\Tests\Controller;
+
+use Payum\Core\Model\Token;
+use Payum\Core\Payum;
+use Payum\Core\Registry\RegistryInterface;
+use Payum\Core\Security\GenericTokenFactoryInterface;
+use Payum\Core\Security\HttpRequestVerifierInterface;
+use Payum\Core\Storage\StorageInterface;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+
+abstract class AbstractControllerTest extends TestCase
+{
+    protected const GATEWAY_NAME = 'theGateway';
+    protected const AFTER_URL = 'http://example.com/theAfterUrl';
+
+    protected $token;
+    protected $httpRequestVerifierMock;
+    protected $gatewayMock;
+    protected $registryMock;
+    protected $payum;
+    protected $request;
+
+    protected function initMocks()
```

Comment:
This can be changed to `setUp` so that it is automatically executed for each test and so that you don't have to call this method each time in the tests